docs/source/topics/data_types.rst

```diff
@@ -4,8 +4,790 @@
+This class can be initialized via the **absolute** path to the file you want to store:
+
+.. code-block:: ipython
+
+  In [1]: SinglefileData = DataFactory('singlefile')
+
+  In [2]: single_file = SinglefileData('/absolute/path/to/file')
+
+The contents of the file in string format can be obtained using the :py:meth:`~aiida.orm.nodes.data.singlefile.SinglefileData.get_content()` method:
+
+.. code-block:: ipython
+
+  In [3]: single_file.get_content()
+  Out[3]: 'The file content'
+
+When storing the node, the filename is stored in the database and the file itself is copied to the repository.
+
+.. _topics:data_types:core:folder:
+
+FolderData
+----------
+
+The :py:class:`~aiida.orm.nodes.data.folder.FolderData` class stores sets of files and folders (including its subfolders).
+To store a complete directory, simply use the ``tree`` keyword:
+
+.. code-block:: ipython
+
+  In [1]: FolderData = DataFactory('folder')
+
+  In [2]: folder = FolderData(tree='/absolute/path/to/directory')
+
+Alternatively, you can construct the node first and then use the various repository methods to add objects from directory and file paths:
+
+.. code-block:: ipython
+
+  In [1]: folder = FolderData()
+
+  In [2]: folder.put_object_from_tree('/absolute/path/to/directory')
+
+  In [3]: folder.put_object_from_file('/absolute/path/to/file1.txt', path='file1.txt')
+
+or from `file-like objects <https://docs.python.org/3/glossary.html#term-file-like-object>`_:
+
+.. code-block:: ipython
+
+  In [4]: folder.put_object_from_filelike(filelike_object, path='file2.txt')
+
+Inversely, the content of the files stored in the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node can be accessed using the :py:class:`~aiida.orm.nodes.node.Node.get_object_content()` method:
+
+.. code-block:: ipython
+
+  In [5]: folder.get_object_content('file1.txt')
+  Out[5]: 'File 1 content\n'
+
+To see the files that are stored in the :py:class:`~aiida.orm.nodes.data.folder.FolderData`, you can use the :py:class:`~aiida.orm.nodes.node.Node.list_object_names()` method:
+
+.. code-block:: ipython
+
+  In [6]: folder.list_object_names()
+  Out[6]: ['subdir', 'file1.txt', 'file2.txt']
+
+In this example, ``subdir`` was a sub directory of ``/absolute/path/to/directory``, whose contents where added above.
+to list the contents of the ``subdir`` directory, you can pass its path to the :py:meth:`~aiida.orm.nodes.node.Node.list_object_names()` method:
+
+.. code-block:: ipython
+
+  In [7]: folder.list_object_names('subdir')
+  Out[7]: ['file3.txt', 'module.py']
+
+The content can once again be shown using the :py:meth:`~aiida.orm.nodes.node.Node.get_object_content()` method by passing the correct path:
+
+.. code-block:: ipython
+
+ In [8]: folder.get_object_content('subdir/file3.txt')
+ Out[8]: 'File 3 content\n'
+
+Since the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node is simply a collection of files, it simply stores these files in the repository
```

Comment:
```suggestion
Since the :py:class:`~aiida.orm.nodes.data.folder.FolderData` node is simply a collection of files, it simply stores these files in the repository.
```